lightning/src/ln/channel.rs

```diff
@@ -3178,40 +3240,109 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 	}
 
-	fn maybe_propose_first_closing_signed<F: Deref>(&mut self, fee_estimator: &F) -> Option<msgs::ClosingSigned>
+	fn calculate_closing_feerate_limits<F: Deref>(&mut self, fee_estimator: &F)
 		where F::Target: FeeEstimator
 	{
-		if !self.is_outbound() || !self.pending_inbound_htlcs.is_empty() || !self.pending_outbound_htlcs.is_empty() ||
-				self.channel_state & (BOTH_SIDES_SHUTDOWN_MASK | ChannelState::AwaitingRemoteRevoke as u32) != BOTH_SIDES_SHUTDOWN_MASK ||
-				self.last_sent_closing_fee.is_some() || self.pending_update_fee.is_some() {
-			return None;
-		}
+		if self.closing_fee_limits.is_some() { return; }
 
+		// Propose a range from our current Background feerate to our Normal feerate plus 10
+		// sat/vbyte. If we fail to come to consensus, we'll have to force-close.
 		let mut proposed_feerate = fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Background);
-		if self.feerate_per_kw > proposed_feerate {
-			proposed_feerate = self.feerate_per_kw;
-		}
+		let mut proposed_max_feerate = if self.is_outbound() {
+				fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Normal)
+			} else {
+				u32::max_value()
+			};
+
+		// The spec requires that (when the channel does not have anchors) we only send absolute
+		// channel fees no greater than the absolute channel fee on the current commitment
```

Comment:
I think the spec could be change here and prepare to splice-out support, where the closing transaction size might be bigger than a commitment one ? Let's avoid to multiply closing pipeline in function of channel type.

Reply:
All of the channel close stuff is based on fees, not feerates. The old code based on feerates being so complicated and brittle I think is good enough reason to move to something fee-based with calculation at start. We'll probably want separate storage for splice negotiation anyway.